src/components/textInput/textInput.tsx

```diff
@@ -62,6 +72,81 @@ export default function TextInput(props: TextInputProps) {
     setMessageText(e.target.value)
   }
 
+  function handleOnFocusToggle() {
+    setIsFocused(!isFocused)
+  }
+
+  function handleToggleSpeechToText() {
+    const microphone = new window.webkitSpeechRecognition()
+
+    microphone.lang = 'en-US'
+
+    if (isRecording) {
+      microphone.stop()
+      setIsEndingRecording(true)
+      setIsRecording(false)
+    } else {
+      microphone.start()
+      setIsRecording(true)
+    }
+
+    microphone.onstart = () => {
+      setIsRecording(true)
+    }
+
+    microphone.onresult = (event: SpeechRecognitionEvent) => {
+      // eslint-disable-next-line no-console
+      console.log('onresult happened')
+      const currentTranscript = event.results[0][0].transcript
+
+      if (messageText.length > 0) {
+        setMessageText(messageText + ' ' + currentTranscript)
+      } else {
+        setMessageText(currentTranscript)
+      }
+
+      setIsRecording(false)
+    }
+
+    microphone.onerror = (event: SpeechRecognitionErrorEvent) => {
+      console.error('Speech recognition error:', event.error)
```

Comment:
Showing the error message on screen might be more helpful.

Reply:
I've added a screenshot to show what this may look like. Let me know what you think.

Reply:
Could you move the error message to the top of the `TextField` so that the positions of error message will be the same for `Input` and `TextInput`? I put the error at top because some buttons are below the `TextField` in the `Input` component. I'm also not sure if we want to rephrase the error message a little bit. The current error message provides enough info, but is not a complete sentence. This could be handled in the future too, if you think it's not that important.

Reply:
I've updated the positioning and error messages.